presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -126,12 +124,18 @@ private ConnectorSplit toIcebergSplit(FileScanTask task)
             }
             else {
                 String partitionValue;
-                if (type.typeId() == FIXED || type.typeId() == BINARY) {
-                    // this is safe because Iceberg PartitionData directly wraps the byte array
-                    partitionValue = new String(((ByteBuffer) value).array(), UTF_8);
-                }
-                else {
-                    partitionValue = value.toString();
+                switch (type.typeId()) {
```

Comment:
Done, and much nicer.  Since in `String.valueOf((Long) value)`, the cast to `Long` is redundant, the former TIME case is correctly handled by the default case, as pointed out by Intellij.